### ISSUE TITLE
Non-finite mass matrix seen in Debug mode

### ISSUE THREAD

**Issue by dougshidong** [dougshidong/PHiLiP#178]
@cyanmcg noticed the following assertion being triggered in Debug mode. This may prevent us from debugging further into the code with gdb since the assertion will simply terminate the code at this location without moving further.

@cyanmcg , could you please post the steps to replicate this?

https://github.com/dougshidong/PHiLiP/blob/master/src/dg/dg.cpp#L2336

![image](https://user-images.githubusercontent.com/9758507/195717575-8241b56c-a2e4-44b6-a7eb-7b342d1bae93.png)

**Comment by cyanmcg** [comment]
As my last check, this bug happened in a special situation.
It can be reproduced in my branch eikonal_model, in the commit:
[eikonal physical source term with hessian components](https://github.com/dougshidong/PHiLiP/commit/8d5ea8cd5521f1131f0fbe32a54218bbf0313370)
To reproduce it, few changes must be done:
https://github.com/dougshidong/PHiLiP/commit/8d5ea8cd5521f1131f0fbe32a54218bbf0313370#:~:text=const%20dealii%3A%3AUpdateFlags%20volume_update_flags%20%3D%20dealii,update_hessians%20%7C%20dealii%3A%3Aupdate_quadrature_points%20%7C%20dealii%3A%3Aupdate_JxW_values%3B

in the above lines, dealii::update_hessians must be removed.

https://github.com/dougshidong/PHiLiP/commit/8d5ea8cd5521f1131f0fbe32a54218bbf0313370#diff-e026c7f12666fa411af87a8eab6d7e20ce2b112986cdfa375242757c3f15c1ec:~:text=InitialConditionFunction_Zero%3Cdim%2Cnstate%2Creal%3E%20%3E%20()%3B-,if%20constexpr%20(dim%3D%3D2%20%26%26%20nstate%3D%3D1)%20%20return%20std,dim%3D%3D2%20%26%26%20nstate%3D%3D1)%20%20return%20std%3A%3Amake_shared%3CInitialConditionFunction_Zero%3Cdim%2Cnstate%2Creal%3E%20%3E%20()%3B,-%7D%20else%20%7B

the InitialConditionFunction_Zero should be used.

The bug can be detected in debug mode from gdb.

**Comment by dougshidong** [comment]
None of the changes you listed should affect the mass matrix calculation. My guess is that it would happen on `master`. What case were you running?

**Comment by cyanmcg** [comment]
You are right. When I tried to reproduce the bug, due to the changes I listed, it seems some other functions called first and the bug does not show up.
Here is the .prm file that I used, may be easier to reproduce the error message.

# Listing of Parameters
# ---------------------
# Number of dimensions
set dimension = 2

set run_type = flow_simulation
set pde_type  = eikonal
set use_weak_form = true

set conv_num_flux = lax_friedrichs
set diss_num_flux = symm_internal_penalty

subsection ODE solver
  # Maximum nonlinear solver iterations
  set nonlinear_max_iterations            = 100

  # Nonlinear solver residual tolerance
  set nonlinear_steady_residual_tolerance = 1e-14

  set initial_time_step = 1000
  set time_step_factor_residual = 20.0
  set time_step_factor_residual_exp = 2.0

  # Print every print_iteration_modulo iterations of the nonlinear solver
  set print_iteration_modulo              = 1

  # Explicit or implicit solverChoices are <explicit|implicit>.
  set ode_solver_type                         = implicit
  
  # set output_final_steady_state_solution_to_file = true
  
  set output_solution_every_x_steps = 4
  
  # set output_solution_every_dt_time_intervals = 0.1
end

subsection linear solver
  subsection gmres options
    set max_iterations = 200
    set linear_residual_tolerance = 1e-4
    set restart_number = 60
  end
end

subsection flow_solver
  set flow_case_type = eikonal_wall
  set steady_state = true
  set steady_state_polynomial_ramping = false
  set poly_degree = 2
  set max_poly_degree_for_adaptation = 7
  subsection grid
    set grid_left_bound  = 0.0
    set grid_right_bound = 1.0
    set number_of_grid_elements_per_dimension = 50
  end
end

**Comment by cyanmcg** [comment]
[2d_eikonal_bounded_wall.txt](https://github.com/dougshidong/PHiLiP/files/9786576/2d_eikonal_bounded_wall.txt)

**Comment by cyanmcg** [comment]
sorry about the pasted version, the display format is not I want

**Comment by cyanmcg** [comment]
Also, I notice my test should in the family of integration_test but not flow_simulation, which I will fix later.
